### PR TITLE
Add new `Plugin*` components types to @wordpress__editor

### PR DESCRIPTION
**Pull request by rafaucau**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - https://make.wordpress.org/core/2024/06/18/editor-unified-extensibility-apis-in-6-6/
  - https://github.com/WordPress/gutenberg/blob/0a7e10db2037cb80e4ab8cda4363ba18d6a3a83f/packages/editor/src/components/plugin-document-setting-panel/index.js
  - https://github.com/WordPress/gutenberg/blob/0a7e10db2037cb80e4ab8cda4363ba18d6a3a83f/packages/editor/src/components/plugin-more-menu-item/index.js
  - https://github.com/WordPress/gutenberg/blob/0a7e10db2037cb80e4ab8cda4363ba18d6a3a83f/packages/editor/src/components/plugin-post-publish-panel/index.js
  - https://github.com/WordPress/gutenberg/blob/0a7e10db2037cb80e4ab8cda4363ba18d6a3a83f/packages/editor/src/components/plugin-post-status-info/index.js
  - https://github.com/WordPress/gutenberg/blob/0a7e10db2037cb80e4ab8cda4363ba18d6a3a83f/packages/editor/src/components/plugin-pre-publish-panel/index.js
  - https://github.com/WordPress/gutenberg/blob/0a7e10db2037cb80e4ab8cda4363ba18d6a3a83f/packages/editor/src/components/plugin-sidebar-more-menu-item/index.js
  - https://github.com/WordPress/gutenberg/blob/0a7e10db2037cb80e4ab8cda4363ba18d6a3a83f/packages/editor/src/components/plugin-sidebar/index.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by gziolo** [comment]
Excellent, I was discussing today with @vaurdan that these newly added components were missing after they were transfered from `@wordpress/edit-post`. Thank you so much for working on it 👍🏻 

By the way, they were deprecated in `@wordpress/edit-post`, but it's still possible to consume them from there. They simply proxy to the new place.

File: types/wordpress__edit-post/package.json [hunk 1]
```diff
@@ -1,15 +1,17 @@
 {
     "private": true,
     "name": "@types/wordpress__edit-post",
-    "version": "7.5.9999",
+    "version": "8.4.9999",
     "projects": [
         "https://github.com/WordPress/gutenberg/tree/master/packages/edit-post/README.md"
     ],
+    "minimumTypeScriptVersion": "5.0",
```
Comment (sheetalkamat) [inline types/wordpress__edit-post/package.json:8]: why is this needed?

**Review by sheetalkamat — DISMISSED** [review]

File: types/wordpress__edit-post/package.json (hunk 1 above)
Comment (rafaucau) [reply types/wordpress__edit-post/package.json:8]: Because this check fails:
https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/10161564814/job/28100341574

Comment (rafaucau) [reply types/wordpress__edit-post/package.json:8]: @sheetalkamat

Comment (rafaucau) [reply types/wordpress__edit-post/package.json:8]: Ping @sheetalkamat

**Review by andrewbranch — APPROVED** [review]

**Comment by rafaucau** [comment]
Ready to merge
